railties/lib/rails/generators/rails/app/templates/config/puma.rb

```diff
@@ -0,0 +1,9 @@
+threads_count = ENV.fetch("MAX_THREADS") { 5 }.to_i
+threads threads_count, threads_count
```

Comment:
You can set it to 0 so that they spin down to nothing and only spin up when used.